### PR TITLE
add scansmangasxyz

### LINKED ISSUE #2571: [Site Request] ScansMangas
**Name of the website**
ScansMangas

**Website urls (examples below)**
- Site: https://scansmangas.xyz/
- Manga List: 
- Manga example: https://scansmangas.xyz/manga/a-returners-magic-should-be-special/
- Chapter online viewer example: https://scansmangas.xyz/scan-a-returners-magic-should-be-special-1/

**Languages**
French

### PR DESCRIPTION
**Pull request by Robonau**
resolves #2571
there probably a better way to getpages

moved the script to the top in WordPressMangastream same way as sinmh

### CODE REVIEW

**Review by 27136773 — COMMENTED** [review]
Looks to me as it would make more sense to overwrite the method `_getPages` and use it with `fetchDOM` instead of introducing the mechanics of configurable scripts...

**Comment by Robonau** [comment]
reverted the WordPressMangastream.mjs changes and overwrite `_getPages` in ScansMangasxyz.mjs

File: src/web/mjs/connectors/ScansMangasxyz.mjs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        this.queryMangas = 'div.bigor > a';
+        this.queryChapters = 'span.lchx.desktop > a';
+        this.queryChaptersTitle = undefined;
+    }
+
+    async _getPages(chapter) {
+        let pagelist = [];
+        const request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        const data = (await this.fetchDOM(request, 'body'))[0];
+        for (let i = 1; i <= parseInt(data.querySelector('div.nav_apb > a:nth-last-of-type(2)').text); i++) {
```
Comment (27136773) [inline src/web/mjs/connectors/ScansMangasxyz.mjs]: May calculate the max number outside of the loop to prevent re-evaluation on each loop cycle.

**Review by 27136773 — APPROVED** [review]
Looks good

File: src/web/mjs/connectors/ScansMangasxyz.mjs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        this.queryMangas = 'div.bigor > a';
+        this.queryChapters = 'span.lchx.desktop > a';
+        this.queryChaptersTitle = undefined;
+    }
+
+    async _getPages(chapter) {
+        let pagelist = [];
+        const request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        const data = (await this.fetchDOM(request, 'body'))[0];
+        const lenloop = parseInt(data.querySelector('div.nav_apb > a:nth-last-of-type(2)').text)
```
Comment (manh21) [inline src/web/mjs/connectors/ScansMangasxyz.mjs]: lint error

**Pull request merged**